Gems/GraphModel/Code/Include/GraphModel/Model/Slot.h

```diff
@@ -110,70 +94,26 @@ namespace GraphModel
         SlotDefinition() = default;
         virtual ~SlotDefinition() = default;
 
-        //! This set of factory functions create a SlotDefinition for each of the valid SlotDirection/SlotType combinations
-        static SlotDefinitionPtr CreateInputData(
-            AZStd::string_view name,
-            AZStd::string_view displayName,
-            DataTypePtr dataType,
-            AZStd::any defaultValue,
-            AZStd::string_view description,
-            ExtendableSlotConfiguration* extendableSlotConfiguration = nullptr,
-            const AZStd::vector<AZStd::string>& enumValues = {},
-            bool visibleOnNode = true,
-            bool editableOnNode = true);
-
-        static SlotDefinitionPtr CreateInputData(
-            AZStd::string_view name,
-            AZStd::string_view displayName,
-            DataTypeList supportedDataTypes,
-            AZStd::any defaultValue,
-            AZStd::string_view description,
-            ExtendableSlotConfiguration* extendableSlotConfiguration = nullptr,
+        SlotDefinition(
+            SlotDirection slotDirection,
+            SlotType slotType,
+            const AZStd::string& name,
+            const AZStd::string& displayName,
+            const AZStd::string& description = {},
+            const DataTypeList& supportedDataTypes = {},
+            const AZStd::any& defaultValue = {},
+            int minimumSlots = {},
+            int maximumSlots = {},
+            const AZStd::string& addButtonLabel = {},
+            const AZStd::string& addButtonTooltip = {},
```

Comment:
The `AZStd::string` parameters should be accepted by value and then moved into their storage.
This prevents an unnecessary copy operation when a string literal or `const char*` is supplied.
```suggestion
            AZStd::string name,
            AZStd::string displayName,
            AZStd::string description = {},
            DataTypeList supportedDataTypes = {},
            AZStd::any defaultValue = {},
            int minimumSlots = {},
            int maximumSlots = {},
            AZStd::string addButtonLabel = {},
            AZStd::string addButtonTooltip = {},
```